File: euler017.hs
word 1 = "one"
word 2 = "two"
word 3 = "three"
word 4 = "four"
word 5 = "five"
word 6 = "six"
word 7 = "seven"
word 8 = "eight"
word 9 = "nine"
word 10 = "ten"
word 11 = "eleven"
word 12 = "twelve"
word 13 = "thirteen"
word 14 = "fourteen"
word 15 = "fifteen"
word 16 = "sixteen"
word 17 = "seventeen"
word 18 = "eighteen"
word 19 = "nineteen"
word 20 = "twenty"
word 30 = "thirty"
word 40 = "forty"
word 50 = "fifty"
word 60 = "sixty"
word 70 = "seventy"
word 80 = "eighty"
word 90 = "ninety"
word 1000 = "onethousand"


word n
  | n `mod` 100 == 0 = word (n `div` 100) ++ "hundred"
  | n > 100 = word (n `div` 100) ++ "hundred" ++ "and" ++ word (n `rem` 100)
  | n >= 20 = word ((n `div` 10) * 10) ++ word (n `rem` 10)

euler17 = sum $ map length [word n | n <- [1..1000]]

File: euler017.py
numbers = {0:'',
           1:'one',
           2:'two',
           3:'three',
           4:'four',
           5:'five',
           6:'six',
           7:'seven',
           8:'eight',
           9:'nine',
           10:'ten',
           11:'eleven',
           12:'twelve',
           13:'thirteen',
           14:'fourteen',
           15:'fifteen',
           16:'sixteen',
           17:'seventeen',
           18:'eighteen',
           19:'nineteen',
           20:'twenty',
           30:'thirty',
           40:'forty',
           50:'fifty',
           60:'sixty',
           70:'seventy',
           80:'eighty',
           90:'ninety',
           100:'hundred'
           }

def write_out(n):
    if n<=20:
        return numbers[n]
    if n<100:
        return numbers[n/10*10] + (('-' + numbers[n%10]) if n%10 else '')
    if n<1000:
        return numbers[n/100] + ' hundred' + \
               ((' and ' + write_out(n%100)) if n%100 else '')

print write_out(5)
print write_out(17)
print write_out(34)
print write_out(50)
print write_out(342)
print write_out(115)
print write_out(200)
print write_out(515)

# the answer for the problem

print sum(len(write_out(n).replace(' ','').replace('-',''))
          for n in xrange(1000)) + len('onethousand')







